### PR TITLE
bugfix: set max_connections sqlx parameter

### LINKED ISSUE #1507: bug: Sqlx(PoolTimedOut) when running multiple indexers
### Description

Reported by @chlenc on Slack.

`fuel-indexer` fails with `Sqlx(PooledTimedOut)` when running multiple indexers.

### Steps to Reproduce

```
cargo run -p fuel-indexer-benchmarks --bin qa -- --network beta-4.fuel.network --blocks 500 --runs 4 --num-additional-indexers 10
```

### PR DESCRIPTION
**Pull request by lostman**
### Description

Fixes #1507.

The default value for `max_connections` in `sqlx` is `10`. From some testing, our number of DB connections per indexer can go up to ~6 and so, running `fuel-indexer` with more than one indexer could lead to failures.

This PR sets the default to `100` and adds a CLI flag to modify it.

Also, a warning will be issued if the set config value exceeds what is set in the backend.

### Testing steps

```
cargo run -p fuel-indexer-benchmarks --bin qa -- --network beta-4.fuel.network --blocks 500 --runs 4 --num-additional-indexers 10
```

Should succeed.

Example: with --num-additional-indexers 10` we get up to `43` connections 
```
run: 1
    runtime:        0.3 minutes
    start block:    0
    end block:      500
    avg memory:     241.3kB
    stdv memory:    0.0kB
    avg cpu:        1.5%
    stdv cpu:       0.0%
    db connections: 43/100
    missing blocks: 0
    blocks/sec:     26.3
    index size:     8.089kB per block
```

### Changelog
* Add --max-db-connections CLI option
* Set the corresponding `sqlx` parameter
* Update QA script to track the number of active connections

### CODE REVIEW

**Review by deekerno — DISMISSED** [review]
LGTM.

**Review by ra0x3 — CHANGES_REQUESTED** [review]
@lostman these manifest files need to be corrected

File: packages/fuel-indexer-benchmarks/src/bin/qa.rs [hunk 1]
```diff
@@ -505,6 +518,45 @@ fn record_mem_usage() -> Result<String, std::io::Error> {
...
+        .arg("--tuples-only")
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn()
+        .unwrap();
+    let output = proc.wait_with_output().unwrap();
+    let output = String::from_utf8(output.stdout).unwrap();
+    let output = output.trim();
+    let output = output.parse::<usize>().unwrap_or(0);
+    println!("Active connections: {output}");
```
Comment (ra0x3) [inline packages/fuel-indexer-benchmarks/src/bin/qa.rs]: We can either set this behind `debug` log level or remove it @lostman - currently running the script right now, this is a bit too much spam

Comment (lostman) [reply packages/fuel-indexer-benchmarks/src/bin/qa.rs]: Agreed. Left it in unnecessarily.

**Review by ra0x3 — APPROVED** [review]
Works as expected. Awesome @lostman

**Review by deekerno — APPROVED** [review]

**Pull request merged**